scripts/sct_straighten_spinalcord.py

```diff
@@ -15,235 +15,242 @@
-                                  'process was run on',
-                      )
-    parser.add_option(name="-r",
-                      type_value="multiple_choice",
-                      description="remove temporary files.",
-                      mandatory=False,
-                      example=['0', '1'],
-                      default_value='1')
-    parser.add_option(name="-v",
-                      type_value="multiple_choice",
-                      description="Verbose. 0: nothing, 1: basic, 2: extended.",
-                      mandatory=False,
-                      example=['0', '1', '2'],
-                      default_value='1')
     return parser
 
 
 # MAIN
 # ==========================================================================================
-def main(args=None):
-    if args is None:
-        args = sys.argv[1:]
-
+def main():
     parser = get_parser()
-    arguments = parser.parse(args)
+    arguments = parser.parse_args(args=None if sys.argv[1:] else ['--help'])
 
     # assigning variables to arguments
-    input_filename = arguments["-i"]
-    centerline_file = arguments["-s"]
+    input_filename = arguments.i
+    centerline_file = arguments.s
 
     sc_straight = SpinalCordStraightener(input_filename, centerline_file)
 
-    if "-dest" in arguments:
+    if arguments.dest is not None:
         sc_straight.use_straight_reference = True
-        sc_straight.centerline_reference_filename = str(arguments["-dest"])
+        sc_straight.centerline_reference_filename = str(arguments.dest)
 
-    if "-ldisc_input" in arguments:
+    if arguments.ldisc_input is not None:
         if not sc_straight.use_straight_reference:
             sct.printv('Warning: discs position are not taken into account if reference is not provided.')
         else:
-            sc_straight.discs_input_filename = str(arguments["-ldisc_input"])
+            sc_straight.discs_input_filename = str(arguments.ldisc_input)
             sc_straight.precision = 4.0
-    if "-ldisc_dest" in arguments:
+    if arguments.ldisc_dest is not None:
         if not sc_straight.use_straight_reference:
             sct.printv('Warning: discs position are not taken into account if reference is not provided.')
         else:
-            sc_straight.discs_ref_filename = str(arguments["-ldisc_dest"])
+            sc_straight.discs_ref_filename = str(arguments.ldisc_dest)
             sc_straight.precision = 4.0
 
     # Handling optional arguments
-    if "-r" in arguments:
-        sc_straight.remove_temp_files = int(arguments["-r"])
-    if "-x" in arguments:
-        sc_straight.interpolation_warp = str(arguments["-x"])
-    if "-o" in arguments:
-        sc_straight.output_filename = str(arguments["-o"])
-    if '-ofolder' in arguments:
-        sc_straight.path_output = arguments['-ofolder']
-    else:
-        sc_straight.path_output = './'
-    path_qc = arguments.get("-qc", None)
-    verbose = int(arguments.get('-v'))
+    if arguments.r is not None:
```

Comment:
it's never None (see how you define it).

the same applies at many other places in your code

Reply:
Is there by chance a preferred default value for arguments.o? Other than that other changes applied in commit [343e697](https://github.com/neuropoly/spinalcordtoolbox/commit/343e697e733bed1064430c73fcc8d77e73d2fc41)